src/trie/merkle_tree.rs

```diff
@@ -177,7 +177,29 @@ impl<H: StarkHash + Send + Sync, DB: BonsaiDatabase, CommitID: Id> MerkleTrees<H
                 // Remove the identifier from the key
                 key_value_pairs
                     .into_iter()
-                    .map(|(key, _value)| key[identifier.len()..].to_vec())
+                    // FIXME: this does not filter out keys values correctly for `HashMapDb` due
+                    // to branches and leafs not being differenciated
+                    .filter(|(key, _value)| key.len() > identifier.len())
+                    .map(|(key, _value)| key[identifier.len() + 1..].to_vec())
```

Comment:
Looks like a perfect usecase for `filter_map`. Don't we have clippy in the CI?

Reply:
Clippy does not appear to be part of the CI currently. Replace `.filter(...).map(...)` by `.filter_map()` in the latest commit.